.clang-format

```diff
@@ -0,0 +1,113 @@
+Language: Cpp
+
+# list of all options:
+# https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+
+Standard: c++17
+AccessModifierOffset: -2
+TabWidth: 2
+IndentWidth: 2
+ConstructorInitializerIndentWidth: 2
+UseTab: Never
+
+# jnl this may be shortened, but this will effect a lot of the formatting
+ColumnLimit: 200
+
+#jnl maybe even 0
+SpacesBeforeTrailingComments: 1
+
+# jnl again, not sure what this does
+ContinuationIndentWidth: 0
```

Comment:
i.e.
```
void foo(int a, 
         int b,
 here -> int c)
...
```

Reply:
no, from https://clang.llvm.org/docs/ClangFormatStyleOptions.html :
```cpp
int i =         //  VeryVeryVeryVeryVeryLongComment
  longFunction( // Again a long comment
    arg);
```

Reply:
yes, you are right, I used the value provided in the config file.
using
```
ContinuationIndentWidth: 2
```
will add two spaces in front of `int c`
```
void foo(int a, 
         int b,
 here -> ++int c)
...

Reply:
this rule is better described in a python linter [here](https://www.flake8rules.com/rules/E125.html)